src/runtime_src/core/tools/common/ReportCu.cpp

```diff
@@ -251,14 +254,23 @@ ReportCu::getPropertyTree20202( const xrt_core::device * _pDevice,
 }
 
 void 
-ReportCu::writeReport( const xrt_core::device* /*_pDevice*/,
+ReportCu::writeReport( const xrt_core::device* _pDevice,
                        const boost::property_tree::ptree& _pt, 
                        const std::vector<std::string>& /*_elementsFilter*/,
                        std::ostream & _output) const
 {
   boost::property_tree::ptree empty_ptree;
+  auto uuid = xrt::uuid(xrt_core::device_query<xrt_core::query::xclbin_uuid>(_pDevice));
   boost::format cuFmt("    %-8s%-30s%-16s%-8s%-8s\n");
 
+#ifndef _WIN32

```

Comment:
Please refrain from using ifdef directives this high in the code. We have worked very hard to remove them from this level. Instead of `uuid_unparse_upper`, can you look into using 
```
/**
   * string_to_UUID(): convert a string to hyphen formatted UUID
   * 
   * Returns: 00000000-0000-0000-0000-000000000000 formatted uuid
   */
  std::string string_to_UUID(std::string str);
```
This function is defined in XButilities.cpp

Maybe
`XBUtillities::string_to_UUID(uuid)` would work